### PR TITLE
Vpupdates merge

### PR DESCRIPTION
**Pull request by yol**
@FernetMenta please check merge commit and renderer registration

Also: Do I have to register/implement a `CProcessInfo`?

### CODE REVIEW

**Comment by FernetMenta** [comment]
> Do I have to register/implement a CProcessInfo?

at least when OpenGL is used. Then the same renderformats and deinterlacing methods are available as we have on X11: https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/Process/X11/ProcessInfoX11.cpp

File: xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp [hunk 1]
```diff
@@ -22,13 +22,35 @@
...
+  if (!CWinSystemWaylandEGLContext::InitWindowSystemEGL(EGL_OPENGL_BIT, EGL_OPENGL_API))
+  {
+    return false;
+  }
+
+  CLinuxRendererGL::Register();  
+  
+#if defined(HAVE_LIBVA)
+  bool general, hevc;
+  CRendererVAAPI::Register(GetVaDisplay(), m_eglContext.m_eglDisplay, general, hevc);
```
Comment (FernetMenta) [inline xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp:46]: We call InitWindowSystem only at startup when the GUI is created. On X11 you can have multiple monitors connected to different graphic cards. This is why I check caps when the GL context is refreshed.

Comment (yol) [reply xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp:46]: what do you mean "refreshed"?

Comment (FernetMenta) [reply xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp:46]: "refresh" is from the old SDL days, sorry. what i mean is that there are situations where the window needs to be recreated.
what do you do when you move a window from a monitor attached to gfx card 1 to a monitor attached to a gfx card 2?

Comment (yol) [reply xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp:46]: I'm honestly not sure whether any Wayland compositor has support for multi-GPU currently. I also vaguely recall that X support for this is not that great and rather hacky?

Also, there's no protocol that could tell you what outputs correspond to which GPU - so even if a compositor one day supports a desktop that spans multiple GPUs, it would have to take care of uploading the rendered buffer you give to it to the right GPU by itself. That a window is being displayed on GPU 2 does not mean it has to be rendered by GPU 2. Depending on the hardware, the GPUs might even be able to share the buffers without copying (like Optimus).

In summary: This does not apply to Wayland, at least for the foreseeable future. There is also no code currently that would destroy/recreate the egl context. It's created in `InitWindowSystem` and destroyed in `DestroyWindowSystem`. I'd have to move that to a separate function first, I guess.

See also: Last paragraph [here](https://nouveau.freedesktop.org/wiki/MultiMonitorDesktop/)

Comment (FernetMenta) [reply xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp:46]: > so even if a compositor one day supports a desktop that spans multiple GPUs

I am not talking about this kind of configuration. On X11 you can have multiple displays (or call it desktop) and each one is connected to a different monitor. 
This configuration is not at all hacky on X11 (at least not with nvidia cards I have tested with). You cannot move the window to the other display but you can select the other one in Kodi's settings. Then the app has to switch.

Comment (yol) [reply xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp:46]: That concept also does not apply to Wayland - Wayland only has "outputs." Switching a full screen window between them is only a Wayland protocol call and not related to the egl context at all.

Comment (FernetMenta) [reply xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp:46]: I am not talking about an egl context but checking hw decoder and rendering capabilities after an output has changed. What happens if you switch the app from an output with an Intel gfx to another output with a nVidia card?

**Review by FernetMenta — APPROVED** [review]
merge commit looks good. i compared the branch against master and did not see any issues due to merge.

**Comment by yol** [comment]
@FernetMenta Thanks for checking the merge commit! I'll merge this to master for now and we can discuss the vaapi capability check some other time.
